### PR TITLE
[KOGITO-3611] Fix "implicit" module dependencies

### PR DESCRIPTION
**Pull request by danielezonca**
https://issues.redhat.com/browse/KOGITO-3611

When there is a failure in one of the core modules of Kogito, for example `kogito-codegen`, all dependent modules should be skipped by maven but this is not happening because a module that contains a `@QuarkusTest` only depends on `kogito-quarkus` and not the compile time component `kogito-quarkus-deployment` that is automatically loaded by Quarkus.
This is a waste of build time and even more important it makes output logs quite noisy because of a lot failed test that was not supposed to run

### Without the fix
```
[INFO] Reactor Summary for Kogito Runtimes 1.0.0-SNAPSHOT:
[INFO] 
[INFO] Kogito Runtimes .................................... SUCCESS [  3.902 s]
[INFO] Kogito Runtimes Build BOM .......................... SUCCESS [  1.591 s]
...
[INFO] Kogito :: Codegen .................................. FAILURE [01:52 min]
...
[INFO] Kogito :: Quarkus Extension :: Deployment .......... SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests ... FAILURE [  9.276 s]
[INFO] Kogito :: Quarkus Extension :: Integration Tests (Hot Reload) SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests (Legacy API) FAILURE [  5.872 s]
[INFO] Kogito :: Maven Archetypes ......................... SUCCESS [  0.012 s]
[INFO] Kogito :: Maven Archetypes :: SpringBoot ........... FAILURE [  3.664 s]
[INFO] Kogito :: Maven Archetypes :: Quarkus .............. FAILURE [ 17.838 s]
[INFO] Kogito :: Integration Tests ........................ SUCCESS [  0.015 s]
[INFO] Kogito :: Integration Tests :: Quarkus :: Rules .... FAILURE [  6.908 s]
[INFO] Kogito :: Integration Tests :: Quarkus :: Processes  FAILURE [ 10.124 s]
[INFO] Kogito :: Integration Tests :: Quarkus :: Decisions  FAILURE [  7.350 s]
[INFO] Kogito :: Integration Tests :: Quarkus :: Predictions FAILURE [  7.101 s]
[INFO] Kogito :: Integration Tests :: SpringBoot .......... SKIPPED
[INFO] Kogito :: Integration Tests :: Maven Plugin ........ SKIPPED
[INFO] Kogito :: Jenkins Spock Tests ...................... SUCCESS [ 24.302 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

### After the fix
```
[INFO] Reactor Summary for Kogito Runtimes 1.0.0-SNAPSHOT:
[INFO] 
[INFO] Kogito Runtimes .................................... SUCCESS [  1.531 s]
[INFO] Kogito Runtimes Build BOM .......................... SUCCESS [  0.479 s]
...
[INFO] Kogito :: Codegen .................................. FAILURE [01:43 min]

...
[INFO] Kogito :: Quarkus Extension :: Deployment .......... SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests ... SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests (Hot Reload) SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests (Legacy API) SKIPPED
[INFO] Kogito :: Maven Archetypes ......................... SUCCESS [  0.008 s]
[INFO] Kogito :: Maven Archetypes :: SpringBoot ........... SKIPPED
[INFO] Kogito :: Maven Archetypes :: Quarkus .............. SKIPPED
[INFO] Kogito :: Integration Tests ........................ SUCCESS [  0.008 s]
[INFO] Kogito :: Integration Tests :: Quarkus :: Rules .... SKIPPED
[INFO] Kogito :: Integration Tests :: Quarkus :: Processes  SKIPPED
[INFO] Kogito :: Integration Tests :: Quarkus :: Decisions  SKIPPED
[INFO] Kogito :: Integration Tests :: Quarkus :: Predictions SKIPPED
[INFO] Kogito :: Integration Tests :: SpringBoot .......... SKIPPED
[INFO] Kogito :: Integration Tests :: Maven Plugin ........ SKIPPED
[INFO] Kogito :: Jenkins Spock Tests ...................... SUCCESS [ 26.543 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by danielezonca** [comment]
Added broken test to verify the behavior ( https://github.com/kiegroup/kogito-runtimes/pull/834/commits/439dc0b01193a797bebad3cd4892b3d861aa1ec8 )

**Comment by danielezonca** [comment]
Ok build of commit https://github.com/kiegroup/kogito-runtimes/pull/834/commits/439dc0b01193a797bebad3cd4892b3d861aa1ec8 fails correctly so removing broken test https://github.com/kiegroup/kogito-runtimes/pull/834/commits/af8014dc8a6565a9e61f307706b44d40aaee8045

```
[INFO] Reactor Summary for Kogito Runtimes 1.0.0-SNAPSHOT:
[INFO] 
[INFO] Kogito Runtimes .................................... SUCCESS [  4.501 s]
[INFO] Kogito Runtimes Build BOM .......................... SUCCESS [  1.900 s]
...
[INFO] Kogito :: Codegen .................................. FAILURE [01:57 min]
...
[INFO] Kogito :: Quarkus Extension :: Deployment .......... SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests ... SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests (Hot Reload) SKIPPED
[INFO] Kogito :: Quarkus Extension :: Integration Tests (Legacy API) SKIPPED
[INFO] Kogito :: Maven Archetypes ......................... SUCCESS [  0.011 s]
[INFO] Kogito :: Maven Archetypes :: SpringBoot ........... SKIPPED
[INFO] Kogito :: Maven Archetypes :: Quarkus .............. SKIPPED
[INFO] Kogito :: Integration Tests ........................ SUCCESS [  0.012 s]
[INFO] Kogito :: Integration Tests :: Quarkus :: Rules .... SKIPPED
[INFO] Kogito :: Integration Tests :: Quarkus :: Processes  SKIPPED
[INFO] Kogito :: Integration Tests :: Quarkus :: Decisions  SKIPPED
[INFO] Kogito :: Integration Tests :: Quarkus :: Predictions SKIPPED
[INFO] Kogito :: Integration Tests :: SpringBoot .......... SKIPPED
[INFO] Kogito :: Integration Tests :: Maven Plugin ........ SKIPPED
[INFO] Kogito :: Jenkins Spock Tests ...................... SUCCESS [ 25.469 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/AddonsConfigTest.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public void allAddonsAreDisabledInDefaultConfiguration() {
     public void addonsAreProperlyActivated() {
         AddonsConfig addonsConfig = new AddonsConfig();
 
-        assertThat(addonsConfig.useMonitoring()).isTrue();
+        assertThat(addonsConfig.useMonitoring()).isFalse();
```
Comment (gitgabrio) [inline kogito-codegen/src/test/java/org/kie/kogito/codegen/AddonsConfigTest.java]: @danielezonca 
Just out of curiosity: what are those two configurations
`addonsConfig.useMonitoring()`
`addonsConfig.withMonitoring(true).useMonitoring()`

, and why they should be different in the test ?

**Review by gitgabrio — APPROVED** [review]
@danielezonca 
LGTM

**Review by radtriste — APPROVED** [review]
lgtm

File: archetypes/kogito-quarkus-archetype/pom.xml [hunk 2]
```diff
@@ -15,6 +15,16 @@
   <name>Kogito :: Maven Archetypes :: Quarkus</name>
   <description>Kogito based on Quarkus runtime Archetype</description>
 
+  <dependencies>
+    <!-- this is used implicitly required by quarkus extension so let's make Maven aware of it -->
+    <dependency>
+      <groupId>org.kie.kogito</groupId>
+      <artifactId>kogito-quarkus-deployment</artifactId>
```
Comment (evacchi) [inline archetypes/kogito-quarkus-archetype/pom.xml:22]: if you want to depend on the extension I think the "proper" way is to just add `kogito-quarkus`. Same for all the other occurrencies

**Review by evacchi — APPROVED** [review]

File: kogito-codegen/src/test/java/org/kie/kogito/codegen/AddonsConfigTest.java (hunk 1 above)
Comment (danielezonca) [reply kogito-codegen/src/test/java/org/kie/kogito/codegen/AddonsConfigTest.java]: @gitgabrio 
As far as I remember `useMonitoring()` is the getter while `withMonitoring(...)` is the setter. Btw I just took a test and made it fail to see the output of the build so I don't know the details of that class

**Pull request merged**